File: apps/hrms-auth/src/main/java/com/cre/hrms/auth/controller/AuthenticationController.java
package com.cre.hrms.auth.controller;

import com.cre.hrms.auth.dto.AuthenticationResponse;
import com.cre.hrms.auth.dto.LoginRequest;
import com.cre.hrms.auth.dto.RegisterRequest;
import com.cre.hrms.auth.service.AuthenticationService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1/auth")
@RequiredArgsConstructor
public class AuthenticationController {

    private final AuthenticationService authenticationService;

    @PostMapping("/register")
    public ResponseEntity<AuthenticationResponse> register(
            @Valid @RequestBody RegisterRequest request
    ) {
        try {
            AuthenticationResponse response = authenticationService.register(request);
            return ResponseEntity.status(HttpStatus.CREATED).body(response);
        } catch (RuntimeException e) {
            throw new RuntimeException("Registration failed: " + e.getMessage());
        }
    }

    @PostMapping("/login")
    public ResponseEntity<AuthenticationResponse> login(
            @Valid @RequestBody LoginRequest request
    ) {
        try {
            AuthenticationResponse response = authenticationService.login(request);
            return ResponseEntity.ok(response);
        } catch (RuntimeException e) {
            throw new RuntimeException("Authentication failed: " + e.getMessage());
        }
    }

    @PostMapping("/refresh")
    public ResponseEntity<AuthenticationResponse> refreshToken(
            @RequestHeader("Authorization") String refreshToken
    ) {
        try {
            if (refreshToken != null && refreshToken.startsWith("Bearer ")) {
                refreshToken = refreshToken.substring(7);
            }
            AuthenticationResponse response = authenticationService.refreshToken(refreshToken);
            return ResponseEntity.ok(response);
        } catch (RuntimeException e) {
            throw new RuntimeException("Token refresh failed: " + e.getMessage());
        }
    }

    @GetMapping("/health")
    public ResponseEntity<String> health() {
        return ResponseEntity.ok("Auth service is running");
    }
}

File: apps/hrms-auth/src/main/kotlin/com/cre/hrms/auth/controller/AuthenticationController.kt
package com.cre.hrms.auth.controller

import com.cre.hrms.auth.service.AuthenticationService
import com.cre.hrms.dto.auth.AuthenticationResponse
import com.cre.hrms.dto.auth.LoginRequest
import com.cre.hrms.dto.auth.RegisterRequest
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v1/auth")
class AuthenticationController(
    private val authenticationService: AuthenticationService
) {

    @PostMapping("/register")
    fun register(
        @Valid @RequestBody request: RegisterRequest
    ): ResponseEntity<AuthenticationResponse> {
        return try {
            val response = authenticationService.register(request)
            ResponseEntity.status(HttpStatus.CREATED).body(response)
        } catch (e: RuntimeException) {
            throw RuntimeException("Registration failed: ${e.message}")
        }
    }

    @PostMapping("/login")
    fun login(
        @Valid @RequestBody request: LoginRequest
    ): ResponseEntity<AuthenticationResponse> {
        return try {
            val response = authenticationService.login(request)
            ResponseEntity.ok(response)
        } catch (e: RuntimeException) {
            throw RuntimeException("Authentication failed: ${e.message}")
        }
    }

    @PostMapping("/refresh")
    fun refreshToken(
        @RequestHeader("Authorization") refreshTokenHeader: String
    ): ResponseEntity<AuthenticationResponse> {
        return try {
            val refreshToken = if (refreshTokenHeader.startsWith("Bearer ")) {
                refreshTokenHeader.substring(7)
            } else {
                refreshTokenHeader
            }
            val response = authenticationService.refreshToken(refreshToken)
            ResponseEntity.ok(response)
        } catch (e: RuntimeException) {
            throw RuntimeException("Token refresh failed: ${e.message}")
        }
    }

    @GetMapping("/health")
    fun health(): ResponseEntity<String> {
        return ResponseEntity.ok("Auth service is running")
    }
}
